### PR TITLE
Feat/marital status

### LINKED ISSUE #42: [Person Form] Add a marital status for a person
It could be an enum with (in French) :
marié
en concubinage
veuf
divorcé
separé
celibataire

### PR DESCRIPTION
**Pull request by jnizet**
On top of #44 

fix #42

### CODE REVIEW

File: frontend/src/app/base-enum-pipe.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { PipeTransform } from '@angular/core';
+
+export class BaseEnumPipe implements PipeTransform {
+
+  constructor(private translations: Array<{key: string; translation: string}>) { }
```
Comment (cexbrayat) [inline frontend/src/app/base-enum-pipe.ts]: Why not use a key/value object here?

File: frontend/src/app/display-gender.pipe.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
-import { Pipe, PipeTransform } from '@angular/core';
+import { Pipe } from '@angular/core';
+import { BaseEnumPipe } from './base-enum-pipe';
+
+export const GENDER_TRANSLATIONS = [
```
Comment (cexbrayat) [inline frontend/src/app/display-gender.pipe.ts]: The keys should be strictly typed (introduce a Gender interface `'male' | 'female' | 'other'`)

File: frontend/src/app/display-marital-status.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { BaseEnumPipe } from './base-enum-pipe';
+
+export const MARITAL_STATUS_TRANSLATIONS = [
```
Comment (cexbrayat) [inline frontend/src/app/display-marital-status.pipe.ts]: The keys should be strictly typed with `MaritalStatus`

**Comment by cexbrayat** [comment]
Arf, nevermind the last two comments, you did what i had in mind in the following PR.

Out of curiosity, I'm still interesting in the first question: why an array with key/value instead of just an object?

File: frontend/src/app/base-enum-pipe.ts (hunk 1 above)
Comment (jnizet) [inline frontend/src/app/base-enum-pipe.ts]: That's what I began with, but the order of keys/entries in an object is undefined, and we want to preserve order of enums generally (the "other" gender, for example, would look weird if it appeared first).

Comment (cexbrayat) [inline frontend/src/app/base-enum-pipe.ts]: 👌

**Pull request merged**